floodlight/io/catapultApi.py

```diff
@@ -0,0 +1,492 @@
+    api_token : str
+        The API token for authentication.
+    activity_id : str
+        The unique identifier for the activity.
+    response : List[Dict]
+        The response containing player data.
+
+    Returns
+    -------
+    List[Dict]
+        A list of response_data for each player, containing sensor data.
+
+    Raises
+    ------
+    APIRequestError
+        If any API request fails with a status code other than 200.
+    """
+    headers = {
+        "accept": "application/json",
+        "authorization": f"Bearer {api_token}",
+        "nulls": "0"
+    }
+
+    player_ids = sorted(set(i["id"] for i in response))
+    response_data_dict_list = []
+
+    response_ok =200
+    for i in player_ids:
+        url = f'activities/{activity_id}/athletes/{i}/sensor'
+        player_data = requests.get(base_url + url, headers=headers)
+        
+        if player_data.status_code == response_ok:
+            response_data_dict_list.append(player_data.json()[0])
+        else:
+            raise APIRequestError(f"API request failed with status code {player_data.status_code}")
+
+    return response_data_dict_list
+
+def get_key_names_from_dict_list(response_data_dict_list: List[Dict]) -> set:
+    """
+    Extracts unique key names from a list of response_data.
+
+    Parameters
+    ----------
+    response_data_dict_list : List[Dict]
+        List of response_data.
+
+    Returns
+    -------
+    set
+        A set of unique key names.
+    """
+    
+    mapping, _, _ = get_mappings()
+
+    recorded_keys = set()
+
+    for athlete_entry in response_data_dict_list:
+        recorded_keys.update(key for key in athlete_entry.keys() if key in mapping)
+        if len(athlete_entry["data"]) != 0:
+            recorded_keys.update(key for key in athlete_entry["data"][0].keys() if key in mapping)
+
+    return recorded_keys
+
+def get_meta_data(response_data_dict_list: List[Dict], coord_format: str = "xy") -> Tuple[Dict[str, Dict[str, List[str]]], int, int, int]:
+    """
+    Extracts metadata from the response_data.
+
+    Parameters
+    ----------
+    response_data_dict_list : List[Dict]
+        List of response_data.
+
+    Returns
+    -------
+    Tuple[Dict[str, Dict[str, List[str]]], int, int, int]
+        - Dictionary mapping group IDs to player identifiers.
+        - Number of frames.
+        - Framerate.
+        - Null timestamp.
+    """
+
+    mapping, reversed_group_mapping, reversed_sensor_mapping = get_mappings()
+    
+    key_names = list(get_key_names_from_dict_list(response_data_dict_list))
+    key_names_mapped = [mapping[i] for i in key_names]
+    sensor_identifier = {"name", "number", "sensor_id", "mapped_id"}
+    key_names_mapped_set = set(key_names_mapped)
+
+    recorded_sensor_identifier = list(key_names_mapped_set & sensor_identifier)
+    sensor_links = {
+        key: index
+        for (index,key) in enumerate(key_names_mapped) #.items()

```

Comment:
I think the inline comment is legacy and can be removed.

Reply:
done